src/entity_resolver/core/graph.py

```diff
@@ -1,16 +1,82 @@
+import re
+import collections
+from nltk.stem import PorterStemmer
+from nltk.corpus import stopwords
+
+
 class Attribute:
 
-    def __init__(self, name, attr_type, value):
+    def __init__(self, name, attr_type, value, deep_clean=True):
         self.name = name
         self.type = attr_type
-        self.value = value
+        # self.adar_amb = None
+        if self.type == 'text':
+            self._tokenize(value)
```

Comment:
I think it is more readable by returning the value in `self._tokenize` and assign it to `self.value` here. The same applies to `self._clean_peson_name` below.